##4006557 fetch all rides

artorious opened the pull request:
#### What does this PR do?

* Fetches all available ride offers

#### Description of Task to be completed?

* Creates and tests an RESTful API endpoint
`GET /api/v1/rides`

#### How the app should be tested
* Set Install, setup and configure the development enviroment.
1. Install [python3](https://www.python.org/download/releases/3.5.2/)
2. [Pipenv & Virtual Environments](http://docs.python-guide.org/en/latest/dev/virtualenvs/)
3. On a  terminal (Linux), Install and set up git `sudo apt install git` 

4.  Clone the repository `git clone https://github.com/artorious/ride_my_way.git`

5. Run `cd ride_my_way`

6. Run `git checkout ft-fetch-all-rides-4006557`

7. In a virutal environment, run `pipenv shell`

8. Test the code `pytest app/tests/`

9. run `FLASK_APP=run.py flask run`

10. Test endpoint on  [Postman](https://www.getpostman.com/) or on terminal, run `curl  --include http://127.0.0.1:5000/api/v1/rides`

#### Any background context you want to provide?
API endpoint  uses data structures  to store data in memory 

#### What are the relevant pivotal tracker stories?
[##4006557](https://www.pivotaltracker.com/epic/show/4006557)


#### Screenshots 

![rides-test](https://user-images.githubusercontent.com/15730750/41959444-ab31ec14-79f5-11e8-9b1a-53cff01a3dba.png)


![post1](https://user-images.githubusercontent.com/15730750/41972707-4cba4b92-7a1b-11e8-8d35-1a41ef7c8886.png)

#### Questions:

artorious committed:
[Feature #158606800] Add function stubs

Adds stubs for models, views and tests.

artorious committed:
[Feature #158606800] Add failing tests

artorious committed:
[Feature #158606800] Add failing tests

artorious committed:
Merge branch 'ft-rides-failing-tests-158606800' of https://github.com/artorious/ride_my_way into ft-rides-failing-tests-158606800

artorious committed:
Merge branch 'ft-rides-failing-tests-158606800' of https://github.com/artorious/ride_my_way into ft-rides-failing-tests-158606800

artorious committed:
Merge branch 'ft-fetch-all-rides-4006557' of https://github.com/artorious/ride_my_way into ft-fetch-all-rides-4006557

artorious committed:
Merge branch 'ft-fetch-all-rides-4006557' of https://github.com/artorious/ride_my_way into ft-fetch-all-rides-4006557

artorious committed:
Merge branch 'ft-fetch-all-rides-4006557' of https://github.com/artorious/ride_my_way into ft-fetch-all-rides-4006557

artorious committed:
[Feature #158606818]Pass tests to fetch rides

Quantum-35 commented on app/models.py:
```diff
@@ -30,11 +30,12 @@ class Rides:
 
     def __init__(self):
-        pass
+        self.all_rides = {}
```
i dont know your case but i think all_rides dictionary should be a class variable i.e move it out of the __init__.py

JoshuaOndieki commented:
Looks good to go

meshack-mbuvi commented:
looks good

Ogollah commented:
Looks good

artorious committed:
[Feature #4006557] define class vaiables

Changes 'all_rides' variable to a class variable
Removes code unused code (part of a later implementation) like
'ride_conter' and 'sample_ride_data'

artorious commented:
Thanks for taking the time @JoshuaOndieki @meshack-mbuvi @Ogollah @Quantum-35 
I've refactored to make the variables class variables.

artorious committed:
Make all_rides variable global

xcixor commented on app/models.py:
```diff
@@ -3,3 +3,38 @@
 """
 
+
+class Rides(dict):
```
As a suggestion instead of inheriting from dict you can have a class or instance variable that holds the rides

artorious committed:
[Feature #158606818] make all_rides var global

Update models.py

artorious committed:
Update models.py

artorious committed:
Merge branch 'ft-fetch-all-rides-4006557' of https://github.com/artorious/ride_my_way into ft-fetch-all-rides-4006557

LWanjiru commented:
@artorious Can you add a screenshot of how this looks on Postman?

artorious committed:
[Feature #158606818] remove global var all_rides

artorious commented:
@LWanjiru I've found my mistake. I made a commit on GitHub to make a variable global which broke the code. I've made adjustments and tested

LWanjiru commented:
Have you set up Travis @artorious ?
